src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoverer.cs

```diff
@@ -140,19 +176,171 @@ private List<PluginFile> GetPluginFiles(CancellationToken cancellationToken)
+                                pluginFiles.Add(pluginFile);
+                            }
+                        }
+                        else
+                        {
+                            // A non DotNet tool plugin file
+                            var state = new Lazy<PluginFileState>(() => _verifier.IsValid(fileInfo.FullName) ? PluginFileState.Valid : PluginFileState.InvalidEmbeddedSignature);
+                            pluginFiles.Add(new PluginFile(fileInfo.FullName, state));
+                        }
+                    }
+                    else if (Directory.Exists(path))
+                    {
+                        var directoryInfo = new DirectoryInfo(path);
+                        var files = directoryInfo.GetFiles("nuget-plugin-*");
+
+                        foreach (var file in files)
+                        {
+                            if (IsValidPluginFile(file))
+                            {
+                                // .NET SDK recently package signature verification for .NET tools, as a result we expect them to be valid.
+                                PluginFile pluginFile = new PluginFile(file.FullName, new Lazy<PluginFileState>(() => PluginFileState.Valid), isDotnetToolsPlugin: true);
+                                pluginFiles.Add(pluginFile);
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    pluginFiles.Add(new PluginFile(path, new Lazy<PluginFileState>(() => PluginFileState.InvalidFilePath)));
+                }
+            }
+
+            return pluginFiles;
+        }
+
+        /// <summary>
+        /// Retrieves .NET tools authentication plugins by searching through directories specified in `PATH` 
+        /// </summary>
+        /// <returns>A list of valid <see cref="PluginFile"/> objects representing the discovered plugins.</returns>
+        internal List<PluginFile> GetPluginsInPATH()
+        {
+            var pluginFiles = new List<PluginFile>();
+            var nugetPluginPaths = _environmentVariableReader.GetEnvironmentVariable("PATH");
+            string[] paths = nugetPluginPaths?.Split(Path.PathSeparator) ?? Array.Empty<string>();
+
+            foreach (var path in paths)
+            {
+                if (PathValidator.IsValidLocalPath(path) || PathValidator.IsValidUncPath(path))
+                {
+                    if (Directory.Exists(path))
+                    {
+                        var directoryInfo = new DirectoryInfo(path);
+                        var files = directoryInfo.GetFiles("nuget-plugin-*");
+
+                        foreach (var file in files)
+                        {
+                            if (IsValidPluginFile(file))
+                            {
+                                // .NET SDK recently package signature verification for .NET tools, as a result we expect them to be valid.
```

Comment:
Those 2 concepts are not related.
This used to check the assembly authenticode signature, not the package.

Reply:
The comment is for `isDotnetToolsPlugin: true` below. So that it does not look like we are using a magic value.

Reply:
My feedback is that the comment is not quite right.

The comment is talking about the package itself, while this plugin validation is supposed to be for the runnable file. 

I think we can just remove the comment. Andy recently removed the authenticode stuff.